### PR TITLE
[INSIGHTS-4] Add Insights user-facing documentation

### PR DESCRIPTION
**Pull request by dinoboy197**
Adding Insights documentation to the dev center. Introductory language taken from [the Product Spec](https://docs.google.com/document/d/1QD9cKyjZmV4Lm1x4jJBnRg4wc9iuC3Il66Uz2FsSOkU/edit?disco=AAAAG5CCoWI). I wrote the remainder of the text. Note that there are some screenshots and code snippets that will be added in [a later piece of work](https://algorithmia.atlassian.net/browse/INSIGHTS-31) once the client library implementation and UX design implementation is finished; that will come before 9/22.

You can play around with it for real in our testing environment - https://insights1.enthalpy.click/developers

Here is the main Insights page, nestled under Algorithmia Enterprise:
 image
image
image
image
image
image

And here are the additions for the Python client library:
image

And R:
image

### CODE REVIEW

**Review by JamesHoover — APPROVED** [review]
This looks great, I wouldn't change a thing

File: _pages/algorithmia-enterprise/insights.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+excerpt: "Sending algorithm metrics to a customer-operated metrics collection system"
+categories: algorithmia-enterprise
+show_related: false
+image:
+  teaser: /icons/algo.svg
+---
+
+Insights is an algorithm metrics pipeline providing you access to your algorithm metrics payload for each algorithm session so you can measure, monitor, and manage your algorithms in production.
+
+With Insights you'll be able to opt-in to exporting your algorithm's operational metrics to external monitoring and alerting tools. This means you’ll be able to monitor your algorithm's execution time and memory consumption as well as capture any model inference metrics you expose from your algorithm's output such as predictions or accuracy. Any metrics you opt to collect will be captured and exported in a payload and, with help from a Platform Administrator, can be connected to your external systems for monitoring and alerting.
```
Comment (eliperelman) [inline _pages/algorithmia-enterprise/insights.md]: Not sure if you wanted to change it for consistency, but there is a stylized apostrophe in the second `you'll`.

File: _pages/algorithmia-enterprise/insights.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Insights currently supports [Python 3.x](../../clients/python) and [R](../../clients/r) algorithms and [Apache Kafka](https://kafka.apache.org/) as a destination for metrics.
+
+### Types of Insights metrics
+
+Insights metrics fall into two categories:
+* Operational metrics
+* Inference-related metrics
+
+**Operational metrics** include (but are not limited to) execution time, memory consumption, request id, algorithm owner, algorithm name, algorithm version, timestamp, etc. When Insights are turned on for a particular algorithm version, these are automatically collected and reported without any additional action necessary on the part of the algorithm author / manager.
```
Comment (eliperelman) [inline _pages/algorithmia-enterprise/insights.md]: Replace "request id" with "request ID".
Replace "author / manager" with "author/manager" or "author or manager".

File: _pages/algorithmia-enterprise/insights.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+**Inference-related metrics** are defined by the algorithm author in algorithm code. Each metric is defined by a title, like `cute_cats_detected` and a numeric value, like `24`. Multiple inference-related metrics can be reported at one time, and the set of metrics reported does not need to be the same from one algorithm execution to another.
```
Comment (eliperelman) [inline _pages/algorithmia-enterprise/insights.md]: Insert comma after `cute_cats_detected`.

File: _pages/algorithmia-enterprise/insights.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+**Screenshot pending UX completion - https://algorithmia.atlassian.net/browse/INSIGHTS-31**
+
+When using the HTTP API or client libraries to publish your algorithm, passing the key value pair `"insights_enabled":true` in the `settings` hash will enable Insights for your algorithm. For example, when using [cURL](../../clients/curl):
+
+{% highlight bash %}
+curl -H 'Authorization: Simple YOUR_API_KEY' https://<algorithmia-cluster-host>/v1/algorithms/<algorithm-owner>/<algorithm-name>/version --data-binary '{"settings":{"insights_enabled":true}}'
+{% endhighlight %}
+
+One you've published your algorithm with Insights enabled, all executions of that specific version of that algorithm will trigger your Insights to be collected and reported to the Kafka broker that your Algorithmia Platform Administrator has set up.
```
Comment (eliperelman) [inline _pages/algorithmia-enterprise/insights.md]: Replace "One" with "Once".
Double "that" usage with "that specific version of that algorithm". Could we change to "that specific version of the algorithm"?

File: _pages/clients/python.md [hunk 5]
```diff
@@ -171,6 +171,18 @@ Now you've seen how to upload a local data file, check if a file exists in a dat
 
 For more methods on how to get a file using the Data API from a data collection go to the [API Specification](http://docs.algorithmia.com/#getting-a-file).
 
+{% if site.enterprise %}
+## Publishing Insights
+
+Inference-related metrics (a feature of [Insights](../../algorithmia-enterprise/insights) can be reported via using the `report_insights` method of the Algorithmia client.
```
Comment (eliperelman) [inline _pages/clients/python.md]: Missing the closing paren for "(a feature of".

File: _pages/clients/python.md [hunk 6]
```diff
@@ -171,6 +171,18 @@ Now you've seen how to upload a local data file, check if a file exists in a dat
...
+
+{% highlight python %}
+# report Insights
```
Comment (eliperelman) [inline _pages/clients/python.md]: Replace "report" with "Report".

File: _pages/clients/r.md [hunk 7]
```diff
@@ -160,6 +160,18 @@ Now you've seen how to upload a local data file, check if a file exists in a dat
 
 For more methods on how to get a file using the Data API from a data collection go to the [API Specification](http://docs.algorithmia.com/#getting-a-file).
 
+{% if site.enterprise %}
+## Publishing Insights
+
+Inference-related metrics (a feature of [Insights](../../algorithmia-enterprise/insights) can be reported via using the `report_insights` method of the Algorithmia client.
```
Comment (eliperelman) [inline _pages/clients/r.md]: Missing the closing paren for "(a feature of".

File: _pages/clients/r.md [hunk 8]
```diff
@@ -160,6 +160,18 @@ Now you've seen how to upload a local data file, check if a file exists in a dat
...
+
+{% highlight r %}
+# report Insights
```
Comment (eliperelman) [inline _pages/clients/r.md]: Replace "report" with "Report".

File: _pages/clients/r.md [hunk 9]
```diff
@@ -160,6 +160,18 @@ Now you've seen how to upload a local data file, check if a file exists in a dat
...
+# TBD R sample code for this - https://algorithmia.atlassian.net/browse/INSIGHTS-31
+client$report_insights(...)
```
Comment (eliperelman) [inline _pages/clients/r.md]: Replace `$` with `.`.

**Review by eliperelman — COMMENTED** [review]
LGTM, just have some grammar nits.

File: _pages/algorithmia-enterprise/insights.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+**Operational metrics** include (but are not limited to) execution time, memory consumption, request id, algorithm owner, algorithm name, algorithm version, timestamp, etc. When Insights are turned on for a particular algorithm version, these are automatically collected and reported without any additional action necessary on the part of the algorithm author / manager.
+
+**Inference-related metrics** are defined by the algorithm author in algorithm code. Each metric is defined by a title, like `cute_cats_detected` and a numeric value, like `24`. Multiple inference-related metrics can be reported at one time, and the set of metrics reported does not need to be the same from one algorithm execution to another.
+
+## One-time administrator setup
+
+The first step in allowing Insights to be collected and reported for algorithms is to have the Algorithmia Platform Administrator connect your cluster to a Kafka broker that your organization operates to receive metrics created by the Algorithmia platform.
+
+Navigate to the _Insights_ administration page, which can be found under the "System Actions" section.
```
Comment (JamesHoover) [inline _pages/algorithmia-enterprise/insights.md]: I am calling the page 'Algorithm Insights' in the UI to be clear it's not platform insights.

Comment (ZachEddy) [inline _pages/algorithmia-enterprise/insights.md]: > _Insights_ ... "System Actions"

When describing which pages to click through, does it make sense to use "quotes" or _italics_ but not both?

Comment (dinoboy197) [reply _pages/algorithmia-enterprise/insights.md]: OK. Is platform insights documented in the dev center so I can read and educate myself about it?

**Review by ZachEddy — APPROVED** [review]

File: _pages/algorithmia-enterprise/insights.md (hunk 10 above)
Comment (JamesHoover) [reply _pages/algorithmia-enterprise/insights.md]: No, not specifically just want the tab to be clear what you're clicking on

Comment (dinoboy197) [reply _pages/algorithmia-enterprise/insights.md]: It probably does, though I was following other pages that follow this convention. Having a "style guide" for the dev center would probably help out here for contributors. @algorithmiaio/ux food for thought

Comment (dinoboy197) [reply _pages/algorithmia-enterprise/insights.md]: Ok, I'll change this wording here. Thanks!

File: _pages/clients/r.md (hunk 9 above)
Comment (dinoboy197) [reply _pages/clients/r.md]: Other method calls documented in this file use `$` as the object / method separator. Should this deviate from those examples?

Comment (eliperelman) [reply _pages/clients/r.md]: Oh my bad, thought this was a typo. :joy:

File: _pages/algorithmia-enterprise/insights.md (hunk 10 above)
Comment (JamesHoover) [reply _pages/algorithmia-enterprise/insights.md]: Yes, let's use quotes. Here is the style guide: https://docs.google.com/document/d/1PPVfgMkX7-EVGLPMhN1E485CAXu9QfhSLKM0lZhnZdU/edit?usp=sharing

This case is not covered so we'll add it

Comment (dinoboy197) [reply _pages/algorithmia-enterprise/insights.md]: Excellent! May I make a recommendation? Update the [Contributing](https://github.com/algorithmiaio/dev-center#contributing) section of the README file in this repository to link to that style guide. Without that reference, there's no way other Algorithmia engineers are going to know that file exists or remember what the link is :)

I'm aware that this repository is open for public contribution, but the practical reality is that the only people who ever edit this repository are Algorithmia engineers :)

**Review by JamesHoover — APPROVED** [review]
🎉

**Comment by dinoboy197** [comment]
@StephanieKim I need final approval from you on this to be able to ship. You can see the finished product live at https://insights1.enthalpy.click/developers. Note that there are some screenshots and code snippets that will be added in a later piece of work once the client library implementation and UX design implementation is finished; that will come before 9/22.

**Review by ZachEddy — APPROVED** [review]
🚢

**Review by stevielkim — CHANGES_REQUESTED** [review]
This looks amazing! Thanks for doing such a great job. I only have one change needed but it needs to happen throughout the guide, but wherever it says "Insights" can we change to "Algorithmia Insights" for branding purposes?

**Review by eliperelman — APPROVED** [review]

**Comment by dinoboy197** [comment]
> I only have one change needed but it needs to happen throughout the guide, but wherever it says "Insights" can we change to "Algorithmia Insights" for branding purposes?

Sure thing - I had that was implicit since we're Algorithmia :) But I realize that maybe marketing is working off of something like "Check out Algorithmia's new product feature, Algorithmia Insights!" Making the change now.

**Comment by dinoboy197** [comment]
> Sure thing - I had that was implicit since we're Algorithmia :) But I realize that maybe marketing is working off of something like "Check out Algorithmia's new product feature, Algorithmia Insights!" Making the change now.

OK @StephanieKim changes added and deployed to https://insights1.enthalpy.click/developers. Please take a look and let me know if you'd like any other changes, or approve if not!

**Pull request closed without merging**